dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java

```diff
@@ -315,6 +316,43 @@ public Response getMessages(@Context HttpServletRequest request,
         return oldLanguagesResource.getMessages(request, i18NForm);
     }
 
+    /**
+     * Takes the requested language and resolves it to a locale
+     * that the Admin console understands
+     * @param language
+     * @return
+     */
+    private Locale resolveAdminLocale(@NotNull String language) {
+        final Locale defaultLocale = APILocator.getCompanyAPI().getDefaultCompany().getLocale();
+        final Locale requestedLocale = ConversionUtils.toLocale(language);
+        final Locale[] locales = LanguageUtil.getAvailableLocales();
+
+        if(null == requestedLocale) {
+            return defaultLocale;
+        }
+        
+        for(int i=0;i<locales.length;i++){
+            if(locales[i].equals(requestedLocale)) {
```

Comment:
why not use an OR here
```
if(locales[i].equals(requestedLocale) ||  locales[i].getLanguage().equalsIgnoreCase(requestedLocale.getLanguage()))

Reply:
Because we we need to check to make sure that there is not a perfect match of country+language before we fall back to just matching against language.